ci/Jenkinsfiles/build.groovy

```diff
@@ -0,0 +1,445 @@
+                    NAMESPACE=${testNamespace} \
+                    DOMAIN=${TEST_SERVICE_DOMAIN_SUFFIX} \
+                    envsubst < ci/mvn/nuxeo-test-${env}.properties~gen > ${HOME}/nuxeo-test-${env}.properties
+                """
+              }
+              // run unit tests for the given environment (see the customEnvironment profile in pom.xml):
+              //   - in an alternative build directory
+              //   - loading some test framework system properties
+              def testCore = env == 'mongodb' ? 'mongodb' : 'vcs'
+              def kafkaOptions = isDev ? '' : "-Pkafka -Dkafka.bootstrap.servers=${kafkaHost}"
+              def mvnCommand = """                 
+                mvn ${MAVEN_ARGS} \
+                  -Dcustom.environment=${env} \
+                  -Dcustom.environment.log.dir=target-${env} \
+                  -Dnuxeo.test.core=${testCore} \
+                  ${kafkaOptions} \
+                  test
+              """
+              sh "${mvnCommand}"
+
+              setGitHubBuildStatus("utests/${env}", "Unit tests - ${env} environment", 'SUCCESS')
+            } catch (err) {
+              echo "${env} unit tests error: ${err}"
+              setGitHubBuildStatus("utests/${env}", "Unit tests - ${env} environment", 'FAILURE')
+              throw err
+            } finally {
+              try {
+                junit allowEmptyResults: true, testResults: "**/target-${env}/surefire-reports/*.xml"
+                if (!isDev) {
+                  archiveKafkaLogs(testNamespace, "${env}-kafka.log")
+                }
+              } finally {
+                echo "${env} unit tests: clean up test namespace"
+                try {
+                  if(!isDev){
+                    helmfileDestroy("${testNamespace}", "${environment}")
+                  }
+                } finally {
+                  // clean up test namespace
+                  sh "kubectl delete namespace ${testNamespace} --ignore-not-found=true"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+String getCurrentNamespace() {
+  container('maven') {
+    return sh(returnStdout: true, script: "kubectl get pod ${NODE_NAME} -ojsonpath='{..namespace}'")
+  }
+}
+
+void archiveKafkaLogs(namespace, logFile) {
+  // don't fail if pod doesn't exist
+  sh "kubectl logs ${TEST_KAFKA_POD_NAME} --namespace=${namespace} > ${logFile} || true"
+  archiveArtifacts allowEmptyArchive: true, artifacts: "${logFile}"
+}
+
+void gitCheckout(String reference, String option = '') {
+  sh """
+    git checkout ${option} ${reference}
+  """
+}
+
+void gitPush(String reference) {
+  sh """
+    git push origin ${reference}
+  """
+}
+
+void gitTag(String tagname, String message) {
+  sh """
+    git tag -a ${tagname} -m "${message}"
+  """
+}
+
+pipeline {
+  agent {
+    label 'jenkins-nuxeo-platform-lts-2021'
+  }
+  options {
+    disableConcurrentBuilds()
```

Comment:
We sould follow what was done for nxueo-lts with https://jira.nuxeo.com/browse/NXP-29543: https://github.com/nuxeo/nuxeo-lts/commit/e8cbf7e0b155f7b797cab33853d7eb9e7429a1aa
So, remove this call to `disableConcurrentBuilds` and add the `abortRunningBuilds` block at the very beigging of the pipeline.

Reply:
Did I add the block in the correct place here?

Reply:
Nope, we want it at the very beginning of the Groovy script, aka after the copyright.
basically, really copy:
```
/*
 * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
 *
 * ...
 */
def abortRunningBuilds() {
  // see https://issues.jenkins.io/browse/JENKINS-43353
  def buildNumber = BUILD_NUMBER as int
  if (buildNumber > 1) {
    milestone(buildNumber - 1)
  }
  milestone(buildNumber)
}
abortRunningBuilds()
```

Reply:
yep